src/System.Data.SqlClient/src/Interop/LocaleInterop.cs

```diff
@@ -87,26 +89,58 @@ private static int LocaleNameToAnsiCodePage(string localeName)
             }
             return s_cachedEncodings.GetOrAdd(lcid, GetDetailsInternal);
         }
+#endif
 
         internal static Encoding GetEncodingForLcid(int lcid)
         {
+#if MANAGED_SNI
+            if (lcid != TdsEnums.DEFAULT_ENGLISH_LOCALE_ID)
+            {
+                throw new NotSupportedException("Only the English-US locale is currently supported on non-Windows systems. Invalid Locale ID: " + lcid);
+            }
+            return Encoding.UTF8;
+#else
             return GetDetailsForLcid(lcid).Item3;
+#endif
         }
 
         internal static int GetCodePageForLcid(int lcid)
         {
+#if MANAGED_SNI
+            if (lcid != TdsEnums.DEFAULT_ENGLISH_LOCALE_ID)
```

Comment:
The MANAGED_SNI references in this file are not appropriate. 
We should create a separate file which compiles on Non-Windows platforms and create the functions in that file.

This approach will hinder the managed sni testing on windows. Especially the localization related aspects will not be tested correctly on windows if we wrap the default locale choosing in MANAGED_SNI if defs.